src/scripts/main.js

```diff
@@ -1 +1,74 @@
 'use strict';
+
+const promise1 = new Promise((resolve, reject) => {
+  const timeout = setTimeout(reject, 3000);
+
+  document.addEventListener('click', () => {
+    clearTimeout(timeout);
+    resolve();
+  });
+});
+const promise2 = new Promise((resolve) => {
+  document.addEventListener('contextmenu', (even) => {
+    even.preventDefault();
+    resolve();
+  });
+
+  document.addEventListener('click', resolve);
+});
+const promise3 = new Promise((resolve, reject) => {
+  let leftClicked = false;
+  let rightClicked = false;
+
+  document.addEventListener('contextmenu', () => {
+    leftClicked = true;
+
+    if (leftClicked && rightClicked) {
+      resolve();
+    }
+  });
+
+  document.addEventListener('click', () => {
```

Comment:
Logical issue: The variable `rightClicked` is being set to `true` on a `click` event, which is misleading. It should be `leftClicked` to match the event type.